### DISCUSSION TITLE
we need to load model once, and keep it there

### DISCUSSION THREAD

**Discussion by falves1** [openai/whisper#2251] [Ideas]
I have a CUDA card with 12 GB of RAM; the large-v3 model takes 10 GB. With the command line command, whisper reloads the model in full every time you transcribe even a small file. If I use Python, the behavior is much more efficient: the model can be loaded once and  pinned to the GPU. I need to achieve that with the command line version. How does it get done?

**Comment by glangford** [comment]
You can list multiple file names as the first argument(s) on the whisper command line, or use wildcarding (eg. *.mp3). whisper will load the model once and iterate through the set of files.

**Reply by glangford** [reply]
https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/transcribe.py#L595

**Reply by falves1** [reply]
will that work in plain bash? like whisper *.wav ?
how do I do monitor a directory so any files copies to it are transcribed without reloading the model, ever? Not using Pyton.
